### PR TITLE
Python: upgrade poetry lock and AI Search document models

### PR DESCRIPTION
**Pull request by seb-jul**
### Motivation and Context

Renamed document models to new name

### Description

Upgrade packages and fix model names

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by moonbox3** [comment]
@seb-jul hmm not sure why the black formatting would be off, but it seems like an easy fix. In the python dir in the terminal, can you please run `black .`? That should fix these [issues](https://github.com/microsoft/semantic-kernel/actions/runs/8070992786/job/22049682631?pr=5223#step:7:22). Almost there! :)

**Comment by seb-jul** [comment]
@moonbox3 yeah 👍🏻 just did it :) for the required files

**Review by eavanvalkenburg — APPROVED** [review]

**Review by moonbox3 — APPROVED** [review]

**Comment by moonbox3** [comment]
@seb-jul there are some[ integration tests](https://github.com/microsoft/semantic-kernel/actions/runs/8071574036/job/22051487073#step:6:134) failing.

**Comment by seb-jul** [comment]
I've identified the root cause as being linked to the latest release of the Azure AI Search SDK version 11.6.0b1.

The key change causing tests to fail is the introduction of a new class, HnswParameters, within the SDK. This class is part of the updates in the azure.search.documents.indexes.models module. For reference, you can find the documentation for this new class here: [HnswParameters Class Documentation](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-search-documents/11.6.0b1/azure.search.documents.indexes.models.html#azure.search.documents.indexes.models.HnswParameters).

File: python/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ async def create_collection(
         else:
             vector_search = VectorSearch(
```
Comment (seb-jul) [inline python/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py:106]: More precisely that's `VectorSearch` class, we need to change the current parameters [VectorSearch Class](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-search-documents/11.6.0b1/azure.search.documents.indexes.models.html#azure.search.documents.indexes.models.VectorSearch)

**Comment by eavanvalkenburg** [comment]
hi @seb-jul could you rebase and run again?

**Comment by eavanvalkenburg** [comment]
@seb-jul parts of this are already in, but I do see something for postgres that are not, could you rebase so we can get the rest in?

**Comment by eavanvalkenburg** [comment]
@seb-jul there are still some postgres things that I don't know, could you have a look after rebasing?

**Pull request closed without merging**